### PR TITLE
ARXIVNG-333 docker runtime for indexing agent

### PR DESCRIPTION
**Pull request by erickpeirson**
This introduces ``Dockerfile-agent``, which relies on some new goodies in ``agent/``. An integration test is in ``tests/integration/test_kinesis_integration.py`` -- it uses docker compose to spin up localstack (for Kinesis, DynamoDB), ES, a metadata stub service, and the agent, and then throws some notifications on the stream.

The metadata stub service is defined in ``Dockerfile-metadata`` and ``tests/stubs/docmeta.py``. This will hopefully make it easier to do more realistic tests around indexing.

### CODE REVIEW

**Comment by erickpeirson** [comment]
This also moves arXiv ID parsing to base, per ARXIVNG-210. Thought I'd knock that out while we had a minor version on base in the pipeline.

File: search/factory.py [hunk 1]
```diff
@@ -21,7 +20,6 @@ def create_ui_web_app() -> Flask:
 
     app = Flask('search')
     app.config.from_pyfile('config.py')
-    app.url_map.converters['arxiv'] = ArXivConverter
```
Comment (JaimieMurdock) [inline search/factory.py]: It seems like this needs a replacement, allows for some nice syntax sugar for `/abs/` urls, but maybe not necessary for search.

Comment (erickpeirson) [reply search/factory.py]: This is now added by base, see https://github.com/cul-it/arxiv-base/blob/master/arxiv/base/__init__.py#L48 ; so available for any app that uses base.

**Comment by erickpeirson** [comment]
@JaimieMurdock Do those comments help a bit?

**Pull request merged**